### ISSUE TITLE
RSA PSS SHA512 Signature Failure

### ISSUE THREAD

**Issue by rafagameiro** [OP-TEE/optee_os#4806]
Greetings,

I was trying to generate a signature using the TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA512 scheme, but during the sign process the TA fails with a TEE_ERROR_BAD_PARAMETERS. I already used this TA to generate signatures using other schemes like TEE_ALG_RSASSA_PKCS1_V1_5_SHA1 and TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA1, with different SHA functions. However, when I specifically tried this cryptographic scheme it failed. Could someone help me figuring out how to solve this error?

Thank you for your time.

Best Regards

**Comment by Emantor** [comment]
Hi,

can you post some minimal example code to reproduce the issue? That would make an investigation much easier.

**Comment by rafagameiro** [comment]
Hi,
I did some research regarding my problem and it seems it was not related with OP-TEE itself, but with the algorithm.
To use the RSA with PSS and SHA512, the key size needs to be at least 2048 otherwise it won't work, that's the reason why my program was not working since I was using a 1024 key size.
Source [1]:
"This results from a condition of PSS (RFC8017, 9.1.1, Def. of emBits), according to which the key/modulus size must be at least the sum of the salt and hash size plus 9 bits. The default salt size is identical to the hash size (RFC8017, A.2.3), resulting in the minimum key sizes 1024/1024/2048 bits for SHA256/384/512 in accordance with your test."

Since I solved my issue, I'll close this

[1] https://stackoverflow.com/questions/66673428/signing-data-is-throwing-exception-for-rsa-sha512-algo-with-key-size-as-512-and
